### PR TITLE
Add need reminders

### PR DESCRIPTION
**Pull request by joelmichael**
We might end up switching to using sidekiq-cron to manage these workers
instead.

Implements https://github.com/OfficeMomsandDads/scheduler/issues/193

### CODE REVIEW

**Comment by joelmichael** [comment]
If this commit with sidekiq-cron looks good, then I can squash down these commits. I initialized the cron job in the remind_worker.rb file, not sure if that's the best way to do it or not. I also added a check to ensure a reminder won't be sent if a new shift was just recently created, because that would have notified them as well.

**Comment by benjaminwood** [comment]
From our call with Sarah:

Needs are sometimes created multiple days in advance. With that in mind, we should send reminders once a day until the day of the need, at which point we can send them hourly (as it is defined now).

**Comment by joelmichael** [comment]
@benjaminwood Take a look at these recent commits and let me know what you think. There's separate logic now for sending both daily and hourly reminders. I added a couple specs for the daily ones.

File: Gemfile [hunk 1]
```diff
@@ -105,4 +105,5 @@ group :test do
   gem 'mutant-rspec',  github: 'mbj/mutant', ref: '90d103dc323eded68a7e80439def069f18b5e990'
   gem 'selenium-webdriver'
   gem 'simplecov', require: false
+  gem 'timecop'
```
Comment (benjaminwood) [inline Gemfile]: Can we use `ActiveSupport::Testing::TimeHelpers`?

https://api.rubyonrails.org/v6.0/classes/ActiveSupport/Testing/TimeHelpers.html

We used to use timecop, but have found it isn't typically needed anymore.

Comment (joelmichael) [reply Gemfile]: Cool, I didn't know about this! Updated

**Comment by joelmichael** [comment]
@benjaminwood I know this has already been approved, but there were a few changes since then and I just want to make sure you feel good about merging it. If so I will.

File: config/initializers/sidekiq.rb [hunk 2]
```diff
@@ -8,3 +8,11 @@
 Sidekiq.configure_server do |config|
   config.redis = Sidekiq::RedisConnection.create(namespace: Rails.env, url: redis_url)
 end
+
+ActiveSupport.on_load(:active_record) do
+  unless Sidekiq::Cron::Job.find('Reminders')
+    Sidekiq::Cron::Job.create(
```
Comment (benjaminwood) [inline config/initializers/sidekiq.rb]: Thoughts on only creating the cron task in certain environments (production only)?

Comment (joelmichael) [reply config/initializers/sidekiq.rb]: That seems fine, I'll update it.

**Pull request merged**